core/src/main/java/org/hisp/dhis/android/core/arch/api/fields/internal/FieldsConverter.java

```diff
@@ -46,45 +44,8 @@ public String convert(Fields fields) throws IOException {
 
         // recursive function which processes
         // properties and builds query string
-        append(builder, (List<Property>) fields.fields());
+        Fields.Companion.append(builder, fields.getFields());
```

Comment:
Ideally we could have a single line with something such as `fields.generateString()` (in the same way we have for FilterConverter)